netty/src/main/java/io/grpc/netty/UdsNameResolver.java

```diff
@@ -58,7 +60,8 @@ private void resolve() {
     List<EquivalentAddressGroup> servers = new ArrayList<>(1);
     servers.add(new EquivalentAddressGroup(new DomainSocketAddress(authority)));
     resolutionResultBuilder.setAddressesOrError(StatusOr.fromValue(servers));
-    listener.onResult2(resolutionResultBuilder.build());
+    args.getSynchronizationContext().execute(() ->
```

Comment:
This was also fine, because start() and refresh() are called from the synchronization context.

Reply:
Removed.